Issue #23 - refined MasterCard pattern

smrobinson opened the pull request:
Refined the existing MasterCard 2-series pattern to only match 2-series BIN numbers in the range 222100 - 272099.

bendrucker commented on src/types.js:
```diff
@@ -27,7 +27,7 @@ exports.dankort = new Type('Dankort', {
 
 exports.masterCard = new Type('MasterCard', {
-  pattern: /^(5[1-5]|2[2-7])\d{14}$/,
+  pattern: /^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$/,
```
No need for the non-capturing flag (`?:`)

bendrucker commented on test.js:
```diff
@@ -110,6 +110,8 @@ test('MasterCard', function (t) {
   var mc = types.masterCard
   t.ok(mc.test('5555555555554444'), 'normal')
   t.ok(mc.test('2223000048400011'), '2 range')
+  t.notOk(mc.test('2723000048400016'), '1st invalid 2 range')
```
Can you add more tests since you added a lot of branches to the regex?

smrobinson commented:
I removed the unneeded capture flag (left over from another app that does capture card type) and added some tests. I also changed the end of the new pattern to use \d instead of [0-9] since that's what is used in other patterns.

smrobinson commented:
@bendrucker Do I need to do anything else before you review this PR again?

bendrucker commented:
You should be good. I've been away but will be home soon and can review this.

smrobinson commented:
Will you be able to review this again soon? Thanks.
